scripts/prove/bin/multi.rs

```diff
@@ -31,17 +33,38 @@ async fn main() -> Result<()> {
     let (l2_start_block, l2_end_block) =
         get_validated_block_range(&data_fetcher, args.start, args.end, DEFAULT_RANGE).await?;
 
-    let host = SingleChainOPSuccinctHost {
-        fetcher: Arc::new(data_fetcher.clone()),
+    let host = {
+        #[cfg(feature = "celestia")]
+        {
+            initialize_celestia_host(Arc::new(data_fetcher.clone()))
+        }
+        #[cfg(not(feature = "celestia"))]
+        {
+            initialize_host(Arc::new(data_fetcher.clone()))
+        }
+    };
+    let host_args = {
+        #[cfg(feature = "celestia")]
+        {
+            host.fetch(
+                l2_start_block,
+                l2_end_block,
+                None,
+                Some(args.safe_db_fallback),
+            )
+            .await?
+        }
+        #[cfg(not(feature = "celestia"))]
+        {
+            host.fetch(
+                l2_start_block,
+                l2_end_block,
+                None,
+                Some(args.safe_db_fallback),
+            )
+            .await?
+        }
     };
```

Comment:
This doesn't need to have two separate branches. When compiled with/without the `celestia` feature flag it will direct to the correct `fetch` function.